awx/api/views/__init__.py

```diff
@@ -1532,6 +1532,18 @@ def post(self, request, *args, **kwargs):
                 return Response(data, status=status.HTTP_400_BAD_REQUEST)
 
             if not role.content_object.organization and not request.user.is_superuser:
+                if request.user.id is user.id and request.data.get('disassociate'):
+                    data = dict(msg=_("You cannot revoke your access to a private credential"))
+                    return Response(data, status=status.HTTP_400_BAD_REQUEST)
+
+                if request.user.id is user.id and not request.data.get('disassociate'):
+                    data = dict(msg=_("You cannot grant yourself access to a private credential"))
+                    return Response(data, status=status.HTTP_400_BAD_REQUEST)
+
+                if request.data.get('disassociate'):
+                    data = dict(msg=_("You cannot revoke another user's access to a private credential"))
+                    return Response(data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Out of all the scenarios, maybe this one should actually be _allowed_. There's still a possibility that someone can get roles to such a credential (after all, superuser can assign them, as you see here). The ability to remove those permissions feels fundamentally like a reasonable expectation.

Reply:
I _think_ that makes sense but let me confirm my understanding a bit: Is the use case to allow users with a private credential to revoke another's access to that credential if it's been granted to them by a sysadmin?

Reply:
yes, but I wouldn't rule out some shenanigans by org admins either. Credentials are something where you probably want a hard ability to revoke access.